### PR TITLE
Add recommendations

### LINKED ISSUE #123: Add build warnings
Because of some incompatible settings, one setting may be ignored. For example requirements checker cannot be used without `stub` to `true`. However I think that bailing out would be a bad UX, instead a warning could be displayed

### PR DESCRIPTION
**Pull request by theofidry**
Closes #123

- Rename `auto-discovery` to `force-autodiscovery`
- Access to the `stdClass` instance properties of the raw configuration via constants for the keys instead of arbitrary strings
- Add recommendations when a configuration setting is set to `null` or the default value. The recommendation being that the setting can be completely ignored
- Remove a few `new InvalidArgumentException()` statements with their respective TODOs in favour of relying on `Assertion` from `baberlei/assert`

### CODE REVIEW

File: doc/configuration.md [hunk 1]
```diff
@@ -183,7 +183,7 @@ or directory from a dev dependency, you can do so by adding it via one of the fo
 binary files, the regular file will take precedence.
 
 
-### Force auto-discovery (`auto-discovery`)
+### Force auto-discovery (`force-autodiscovery`)
 
 The `auto-discovery` (`bool` default `false`) setting forces Box to attempt to find which files to include even though
```
Comment (keradus) [inline doc/configuration.md]: shall be renamed as well?

change of input name (CLI param, param in config file) is a BC breaker

File: src/Configuration.php [hunk 2]
```diff
@@ -96,6 +98,47 @@
     private const PHP_SCOPER_CONFIG = 'scoper.inc.php';
     private const DEFAULT_SIGNING_ALGORITHM = Phar::SHA1;
 
+    private const ALGORITHM_KEY = 'algorithm';
```
Comment (keradus) [inline src/Configuration.php]: I would suggest to put them into separated, non-instanceable class, or use prefix (instead of suffix) for easier distinguishability

File: tests/Console/Command/CompileTest.php [hunk 3]
```diff
@@ -605,7 +602,7 @@ public function test_it_can_build_a_PHAR_with_complete_mapping(): void
 ? Dumping the Composer autoloader
 ? Removing the Composer dump artefacts
 ? No compression
-? Setting file permissions to 0755
+? Setting file permissions to 0754
```
Comment (keradus) [inline tests/Console/Command/CompileTest.php]: out of curiosity,
why the change ?

File: src/Configuration.php (hunk 2 above)
Comment (theofidry) [reply src/Configuration.php]: I still have mixed feelings about it as well to be honest, but for now I think I prefer to keep it as it is
